devolutions-gateway/src/api/monitoring.rs

```diff
@@ -0,0 +1,222 @@
+use crate::DgwState;
+use crate::http::HttpError;
+use axum::{Json, Router, extract, routing};
+use network_monitor;
+use time::OffsetDateTime;
+
+pub fn make_router<S>(state: DgwState) -> Router<S> {
+    let router = Router::new()
+        .route("/config", routing::post(handle_set_monitoring_config))
+        .route("/log/drain", routing::post(handle_drain_log));
+
+    router.with_state(state)
+}
+
+/// Replaces the existing monitoring config with the one provided in the body.
+/// This request will immediately start any new monitors, and will stop
+/// currently active monitors that are no longer in the config.
+///
+/// The configuration is not persisted across restarts.
```

Comment:
question: By the way, I kind of remember you talked about that in the past, but I’m not entirely sure: is it really fine to not persist the configuration? I guess the idea is that if the Gateway is restarted, the monitor order must be reissued by DVLS?

Reply:
That's the idea, given that the controlling server needs to be in regular contact with gateway, we might as well keep it stateless. But I've been meaning to ask you: does DVLS have an explicit way to know when the gateway comes online? It's not clear to me how these "Subscribers" work. If not I would have to rethink the approach.